ntp-daemon/src/main.rs

```diff
@@ -4,23 +4,32 @@ use clap::Parser;
 use ntp_daemon::config::{CmdArgs, Config};
 use ntp_daemon::Peers;
 use std::{error::Error, sync::Arc};
+use tracing::error;
 use tracing_subscriber::EnvFilter;
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error>> {
     let args = CmdArgs::parse();
     let has_log_override = args.log_filter.is_some();
+    let has_format_override = args.log_format.is_some();
     let log_filter = args.log_filter.unwrap_or_else(|| EnvFilter::new("info"));
 
     // Setup some basic tracing now so we are able
     // to log errors when loading the full configuration.
-    let finish_tracing_init = ntp_daemon::tracing::init(log_filter);
+    let finish_tracing_init =
+        ntp_daemon::tracing::init(log_filter, args.log_format.unwrap_or_default());
 
-    let mut config = Config::from_args(args.config, args.peers).await?;
+    let mut config = match Config::from_args(args.config, args.peers).await {
+        Ok(c) => c,
+        Err(e) => {
+            error!("There was an error in the configuration: {}", e);
+            std::process::exit(78);
```

Comment:
why 78? maybe document the magic number?

Reply:
Hmm I shouldn't have pushed that code with this pull request, I'll split it out into a separate one. The number 78 is from `/usr/include/sysexits.h` btw, but that should definitely be documented.